### PR TITLE
Lab4 and 5: Numeronyms and Stringer

### PR DESCRIPTION
**Pull request by amitkrishna85**
Fixes #78  

#### Changes proposed in this PR:

- Add the numeronyms code and unit test
- Add the Stringer code and unit test

### CODE REVIEW

File: 04_numeronym/amitkrishna/numeronym.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, numeronyms("accessibility", "Kubernetes", "abc"))
+}
+
+func numeronyms(vals ...string) []string {
+	var slice []string
```
Comment (juliaogris) [inline 04_numeronym/amitkrishna/numeronym.go]: This is fine, but if you really care about runtime you could optimise by preallocating the size of `numeronyms`:
```go
s := make([]string, len(vals))
for i, val := range vals {
	s[i] =  numeronym(val)
}
```

File: 04_numeronym/amitkrishna/numeronym_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+	"github.com/stretchr/testify/require"
+)
+
+var numeronymInputData = []struct {
+	n        []string // input
+	expected []string // expected result
+}{
+	{[]string{"accessibility", "Kubernetes", "abc"}, []string{"a11y", "K8s", "abc"}},
+	{[]string{"a", "b", "abc", "abcd"}, []string{"a", "b", "abc", "a2d"}},
```
Comment (juliaogris) [inline 04_numeronym/amitkrishna/numeronym_test.go:15]: Add unicode tests, test empty string, spaces.

**Review by juliaogris — COMMENTED** [review]
Looks good, pretty close!

File: 04_numeronym/amitkrishna/numeronym.go (hunk 1 above)
Comment (amitkrishna85) [reply 04_numeronym/amitkrishna/numeronym.go]: Allocated the size.

File: 04_numeronym/amitkrishna/numeronym_test.go (hunk 2 above)
Comment (amitkrishna85) [reply 04_numeronym/amitkrishna/numeronym_test.go:15]: Added the test

**Review by juliaogris — APPROVED** [review]
LGTM 👍 

If you can next time fold commits into just one or two.

**Pull request merged**